numba/tests/test_parfors.py

```diff
@@ -2409,7 +2411,12 @@ def run_pass(self, state):
                             # point it needs to be a set so e.g. set.difference
                             # can be computed, this therefore creates an error
                             # in the right location.
-                            stmt.races = []
+                            class Broken(list):
+
+                                def difference(self, other):
+                                    raise errors.LoweringError(BROKEN_MSG)
+
+                            stmt.races = Broken()
```

Comment:
Why is test necessary? When I undo the changes to this test, i can't make it fail either error style.

Reply:
Is the test running or skipping? The parfors tests need the subprocess test harness to actually run, else they "run" but just skip internally. Invocation I'm using is:
```
SUBPROC_TEST=1 NUMBA_CAPTURED_ERRORS='new_style' ./runtests.py numba.tests.test_parfors.TestParforsMisc.test_no_state_change_in_gufunc_lowering_on_error
```
which gives:
```
umba/parfors/parfor_lowering.py", line 989, in _create_gufunc_for_parfor_body
    races = races.difference(set(parfor_redvars))
AttributeError: 'list' object has no attribute 'difference'
```
without this change. What this change does is regularise the failure.

Reply:
Ok, i see the point on regularizing the failure.

(btw, i am amazed at my inability to type at 7pm. the original comment is incomprehensible 🤦 .)